generator/template.go

```diff
@@ -195,6 +195,97 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+type RequestPayload = Record<string, unknown>;
+type Primitive = string | boolean | number;
+
+/**
+ * Checks if given value is of a primitive type
+ * @param  {unknown} value
+ * @return {boolean}
+ */
+function isPrimitive(value: unknown): boolean {
+  return (
+    typeof value === "string" ||
+    typeof value === "number" ||
+    typeof value === "boolean"
+  );
+}
+
+/**
+ * Flattens a deeply nested request payload and returns an object
+ * with only primitive values and non-empty array of primitive values
+ * @param  {RequestPayload} requestPayload
+ * @param  {String} path
+ * @return {Record<string, Primitive | Array<Primitive>}
+ */
+function flattenRequestPayload<T extends RequestPayload>(
+  requestPayload: T,
+  path: string = ""
+): Record<string, Primitive | Array<Primitive>> {
+  return Object.keys(requestPayload).reduce(
+    (acc: T, key: string): T => {
```

Comment:
there's also case where the key could be assigned to null and it will present in the keys because it's set.

`Object.keys({a: null}).forEach(k => console.log("key", k))` => `key a`

`Object.keys({a: undefined}).forEach(k => console.log("key", k))` => `key a`

You might want to filter out the null/undefined case here.

Reply:
^^ also could be added to the test case you have

Reply:
in fact, wouldn't the falsy cases the default value anyway? can you verify?  that might also save the length of the url but it will lose some readability on the URL. <- as in, `"I've set the value to 0/false, but why it doesn't show up in the url?`. It might just worth a sentense in the readme for this.

Reply:
The function `isPlainObj.js` will ensure that undefined and null are filtered out. I will add a test case to check this as well 🙇🏼

Reply:
I will add tests to verify falsy cases such as a `0/false`. 👍🏼

Reply:
oh, right. I see. ah. all good. yeah. 0 & false case then